### PR TITLE
added pre-commit hooks

### LINKED ISSUE #169: Need pre-commit hooks
I see in a lot of PR there are flake8 and isort errors. These could be easily avoided if there pre-commit hooks are used. Also we can use black as code formatter and switch to github actions.

### PR DESCRIPTION
**Pull request by jdvala**
Unfortunately, pre-commit hooks does not support python3.4, so I had to drop support for python3.4. Its upto @kvesteri to make a decision. closes #169

### CODE REVIEW

**Comment by kvesteri** [comment]
Why did you change single-quotes to double-quotes?

**Comment by jdvala** [comment]
Sorry about that, so when I started implementing the pre-commits, I included black formatter but it was not supported on python3.4 as well as 3.5 so had to remove it. It was black that formatted the single quotes to double quotes. I can undo the changes if its necessary?

**Comment by kvesteri** [comment]
Please do, I personally like single quotes more and see no reason for changing those.

**Comment by jdvala** [comment]
I have reverted back the changes that black had made. Please tell me if there is anything out of place, I will be happy to change it.

**Comment by jdvala** [comment]
@kvesteri Could you may be please have a look

**Comment by mondeja** [comment]
You forgot to revert them in `setup.py`.

File: setup.py [hunk 1]
```diff
@@ -6,30 +6,16 @@
...
-import re
 import sys
 
+from setuptools import find_packages, setup
 
 PY3 = sys.version_info[0] == 3
 HERE = os.path.dirname(os.path.abspath(__file__))
 
-
-def get_version():
```
Comment (kvesteri) [inline setup.py:19]: Why did you remove this function? I find this function really convenient in a sense that I don't have to remember to update the version number in two different places. If this is removed then I'd like to have a viable alternative for this rather than hard-coding the version number in this file as well.

**Review by kvesteri — CHANGES_REQUESTED** [review]

**Comment by jdvala** [comment]
Reverted back! Also @kvesteri why we are still not using Github actions?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[isort]
+known_first_party=sqlalchemy_utils,tests
+line_length=88
+multi_line_output=3
+order_by_type=false
+include_trailing_comma=True
+
+[flake8]
+ignore = D10,E203,E501,W503,E231,E265,W504,E122,E121
```
Comment (kvesteri) [inline setup.cfg:9]: Is there a reason for adding all these ignores?

**Review by kvesteri — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 2 above)
Comment (jdvala) [reply setup.cfg:9]: Some of these were already set, and there were already a lot of violation for flake8 so for the most common ones I added it to ignore. If I remove them then I would have to do a lot of editing to code which I am not too sure if you would like it.

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[isort]
+known_first_party=sqlalchemy_utils,tests
+line_length=88
```
[the remark this answers is not in the document]
Comment (jdvala) [reply setup.cfg:3]: Sure, I will change it back.

File: setup.cfg (hunk 2 above)
Comment (jdvala) [reply setup.cfg:9]: @kvesteri could you please confirm if these ignores are ok? I can then submit my PR. Thanks
